packages/mobile/src/components/block-messages-drawer/BlockMessagesDrawer.tsx

```diff
@@ -121,15 +125,22 @@ export const BlockMessagesDrawer = () => {
         dispatch(createChat({ userIds: [userId] }))
       }
     } else {
-      dispatch(blockUser({ userId }))
+      dispatch(isReportAbuse ? reportUser({ userId }) : blockUser({ userId }))
```

Comment:
Could also always dispatch `blockUser` and separately optionally dispatch `reportUser`?